src/dbms/transaction.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include "storage/v2/config.hpp"
+
+namespace memgraph::dbms {
+struct SystemTransaction {
+  struct Delta {
```

Comment:
Unlike data Deltas, which represent undo action (so it can rewind to previous timestamp), this System change is just an operation deferred to commit time.

Reply:
We use the term Delta for both MVCC(undo action) and WAL(do action). Lets keep it as is for now.